ios/engine/KMEI/KeymanEngine/Classes/KeymanWebViewController.swift

```diff
@@ -327,6 +331,38 @@ extension KeymanWebViewController: WKScriptMessageHandler {
     // kmw adds w/2 to x.
     return CGRect(x: x - w / 2.0, y: y, width: w, height: h)
   }
+  
+  public func beep(_ keymanWeb: KeymanWebViewController) {
+    let vibrationSupport = Manager.shared.vibrationSupportLevel
+    let kSystemSoundID_LightVibrate: SystemSoundID = 1519
+
+    if vibrationSupport == .none {
+      // TODO:  Find something we can do visually and/or audibly to provide feedback.
```

Comment:
We have the ability to play sounds ... so we could play a mac "crash" sound ;-) Or something slightly less scary, like a yeti screaming.